lib/matplotlib/animation.py

```diff
@@ -288,17 +288,17 @@ def __init__(self, fps=5, codec=None, bitrate=None, extra_args=None,
         self.frame_format = 'rgba'
 
         if codec is None:
-            self.codec = rcParams['animation.codec']
+            self.codec = mpl.rcParams['animation.codec']
```

Comment:
these changes are necessary to make animation importable from rcsetup.

Reply:
Semi-OT: How can we reasonably handle the import hierarchy? Modules want to import `rcsetup` to have access to `rcParams` but `rcsetup` also wants to import some modules to read valid parameter values. I'm a bit concerned we'll run into cyclic imports sooner or later.

Reply:
In general I think doing `import matplotlib as mpl` and then accessing the rcParams as `mpl.rcParams` is good enough (as long as that access doesn't happen at module scope).

Reply:
We then currently have the asymmetry, that some modules use that pattern while others use `from matplotlib import rcParams`. Should we switch all cases to `mpl.rcParams`? Importing a single variable into a module is anyway not too nice.

Reply:
I think they should all be switched to use mpl.rcParams to help with the circular import problem, but I don't think needs to be done *now*.

Reply:
Agreed.